src/sentry/utils/managers.py

```diff
@@ -1,70 +1,112 @@
 """
-sentry.utils.db
-~~~~~~~~~~~~~~~
+Original copyright:
 
 :copyright: (c) 2010-2014 by the Sentry Team, see AUTHORS for more details.
 :license: BSD, see LICENSE for more details.
 """
-from __future__ import absolute_import, print_function
 
+from __future__ import absolute_import
+import six
 import logging
 
+logger = logging.getLogger(__name__)
+
+
+class LoadException(Exception):
+    pass
+
+
+class ImportException(LoadException):
+    pass
+
+
+class InitializeException(LoadException):
+    pass
+
+
+LAZY_MARKER = object()
+
 
 class InstanceManager(object):
-    def __init__(self, class_list=None, instances=True):
-        if class_list is None:
-            class_list = []
-        self.instances = instances
-        self.update(class_list)
+
+    ImportException = ImportException
+
+    InitializeException = InitializeException
+
```

Comment:
I have seen you do this, but I don't understand why? Just out of curiosity, could you explain?

Reply:
Right. In this case I think I might actually have just declared the classes in the InstanceManager itself, and I think it's a bit strange that I didn't. The idea is basically that the exception does not need to be imported, so the user can just do:
```
try:
...
except InstanceManager.ImportException
```

and it's then clear that it's that ImportException the InstanceManager throws (not some other import exception).

Django does this a lot and I think I'm influenced by that. As with so many design patterns, I both like it and I don't. I think it may be clearer to just import the exception as is more usual.

However, I think you might mainly have seen me do this up until now in the plugin-developer-facing code (?) There I think there is an argument for that the user doesn't have to import too much, as it might be a bit easier to get into.

I'm of course fine with changing this if it causes confusion or distress down the line :) But for now I hope it's OK.

Reply:
Thank you for the explanation. I like the argument about the plugin dev not having to import as much. And it's absolutely fine, I'm just educating myself.